src/main/java/org/broadinstitute/hellbender/Main.java

```diff
@@ -91,6 +93,36 @@ protected static void printDecoratedExceptionMessage(final PrintStream ps, final
         return packageList;
     }
 
+
+    /**
+     * Reads from the given command-line arguments, pulls out configuration options,
+     * and initializes the configuration for this instance of Main.
+     * Returns the list of arguments without the config file option and config file path.
```

Comment:
As we talked about in person I'll make the change to add the config file path to CommandLineProgram, but leave parsing in the Main so that it can be read in time for later programs.